pkg/ccl/streamingccl/streamproducer/producer_job.go

```diff
@@ -187,10 +205,30 @@ func updateReplicationStreamProgress(
 			}
 			return nil
 		})
+
+	if jobs.HasJobNotFoundError(err) || testutils.IsError(err, "not found in system.jobs table") {
```

Comment:
do we need to depend on testutils?  Are we getting that error message?
Regardless, I think HasJobNotFoundError should handle the case and we shouldn't depend on the testutil package.